internal/metrics/listener.go

```diff
@@ -148,16 +150,52 @@ func getEnhancedMetricsTags(ctx context.Context) []string {
 	isColdStart := ctx.Value("cold_start")
 
 	if lc, ok := lambdacontext.FromContext(ctx); ok {
-		// ex: arn:aws:lambda:us-east-1:123497558138:function:golang-layer
+		// ex: aS
 		splitArn := strings.Split(lc.InvokedFunctionArn, ":")
+		var alias string
+		var executedversion string
+		var resource string
 
 		functionName := fmt.Sprintf("functionname:%s", lambdacontext.FunctionName)
 		region := fmt.Sprintf("region:%s", splitArn[3])
 		accountId := fmt.Sprintf("account_id:%s", splitArn[4])
 		memorySize := fmt.Sprintf("memorysize:%d", lambdacontext.MemoryLimitInMB)
 		coldStart := fmt.Sprintf("cold_start:%t", isColdStart.(bool))
-		return []string{functionName, region, accountId, memorySize, coldStart}
+
+		tags := []string{functionName, region, accountId, memorySize, coldStart}
+		if len(splitArn) > 7 {
+			alias = splitArn[7]
+		}
+		// Check we have an arn
+		if alias != "" {
```

Comment:
This whole block is kinda confusing. Took me a while to figure out what's going on here. It might be easier to reformat it similar to this:

https://play.golang.org/p/nl9APlB7v5k

Reply:
Updated with a switch statement instead. Thanks for the suggestion